test/types/fetch.test-d.ts

```diff
@@ -0,0 +1,137 @@
+expectType<Promise<Response>>(fetch('https://example.com', requestInit))
+expectType<Promise<Response>>(fetch(new URL('https://example.com'), requestInit))
+
+expectType<Response>(new Response())
+expectType<Response>(new Response(null))
+expectType<Response>(new Response('string'))
+expectType<Response>(new Response(new Blob([])))
+expectType<Response>(new Response(new FormData()))
+expectType<Response>(new Response(new Int8Array()))
+expectType<Response>(new Response(new Uint8Array()))
+expectType<Response>(new Response(new Uint8ClampedArray()))
+expectType<Response>(new Response(new Int16Array()))
+expectType<Response>(new Response(new Uint16Array()))
+expectType<Response>(new Response(new Int32Array()))
+expectType<Response>(new Response(new Uint32Array()))
+expectType<Response>(new Response(new Float32Array()))
+expectType<Response>(new Response(new Float64Array()))
+expectType<Response>(new Response(new BigInt64Array()))
+expectType<Response>(new Response(new BigUint64Array()))
+expectType<Response>(new Response(new ArrayBuffer(0)))
+expectType<Response>(new Response(null, responseInit))
+expectType<Response>(new Response('string', responseInit))
+expectType<Response>(new Response(new Blob([]), responseInit))
+expectType<Response>(new Response(new FormData(), responseInit))
+expectType<Response>(new Response(new Int8Array(), responseInit))
+expectType<Response>(new Response(new Uint8Array(), responseInit))
+expectType<Response>(new Response(new Uint8ClampedArray(), responseInit))
+expectType<Response>(new Response(new Int16Array(), responseInit))
+expectType<Response>(new Response(new Uint16Array(), responseInit))
+expectType<Response>(new Response(new Int32Array(), responseInit))
+expectType<Response>(new Response(new Uint32Array(), responseInit))
+expectType<Response>(new Response(new Float32Array(), responseInit))
+expectType<Response>(new Response(new Float64Array(), responseInit))
+expectType<Response>(new Response(new BigInt64Array(), responseInit))
+expectType<Response>(new Response(new BigUint64Array(), responseInit))
+expectType<Response>(new Response(new ArrayBuffer(0), responseInit))
+expectType<Response>(Response.error())
+expectType<Response>(Response.redirect('https://example.com', NaN))
+
+expectType<void>(headers.append('key', 'value'))
+expectType<void>(headers.delete('key'))
+expectType<string | null>(headers.get('key'))
+expectType<boolean>(headers.has('key'))
+expectType<void>(headers.set('key', 'value'))
+expectType<IterableIterator<string>>(headers.keys())
+expectType<IterableIterator<string>>(headers.values())
+expectType<IterableIterator<[string, string]>>(headers.entries())
+
+expectType<RequestCache>(request.cache)
+expectType<RequestCredentials>(request.credentials)
+expectType<RequestDestination>(request.destination)
+expectType<Headers>(request.headers)
+expectType<string>(request.integrity)
+expectType<string>(request.method)
+expectType<RequestMode>(request.mode)
+expectType<string>(request.redirect)
```

Comment:
Can you strictly type this? https://developer.mozilla.org/en-US/docs/Web/API/Request/redirect